c10/core/SymInt.h

```diff
@@ -39,18 +39,13 @@ class C10_API SymInt {
     return (MASK & static_cast<uint64_t>(this->data_)) == IS_SYM;
   }
 
-  bool operator==(const SymInt& p2) const {
-    return data_ == p2.data_;
-  }
-
-  bool operator!=(const SymInt& p2) const {
-    return data_ != p2.data_;
-  }
-
   SymInt operator+(SymInt sci) const;
   SymInt operator*(SymInt sci) const;
   bool operator<(SymInt sci) const;
   void operator*=(SymInt sci);
+  // Perhaps these should return symbolic bools?
```

Comment:
would be nice, but having the immediately guard is probably not too bad for now.